openupgrade_scripts/scripts/hr/16.0.1.1/post-migration.py

```diff
@@ -0,0 +1,186 @@
+            "relation_source_field": AsIs(relation_source_field),
+            "relation_comodel_field": AsIs(relation_comodel_field),
+        },
+    )
+
+
+def create_work_contact(env):
+    """Create work_contact_id for model hr.employee.base"""
+    # Get employee
+    openupgrade.logged_query(
+        env.cr,
+        """
+        SELECT id, name, work_email, mobile_phone, company_id
+        FROM hr_employee
+        """,
+    )
+    for employee_values in env.cr.fetchall():
+        # Create partner
+        openupgrade.logged_query(
+            env.cr,
+            """
+            INSERT INTO res_partner (name, email, mobile, company_id)
+            VALUES (%s, %s, %s, %s)
+            RETURNING id
+            """,
+            (
+                employee_values[1],
+                employee_values[2],
+                employee_values[3],
+                employee_values[4],
+            ),
+        )
+        partner_values = env.cr.fetchall()
+        # Update employee
+        openupgrade.logged_query(
+            env.cr,
+            "UPDATE hr_employee SET work_contact_id = %s WHERE id = %s",
+            (partner_values[0][0], employee_values[0]),
+        )
+        # Create image
+        openupgrade.logged_query(
+            env.cr,
+            """
+            SELECT
+                name, description, res_model, res_field, res_id,
+                company_id, type, url, public, db_datas, store_fname,
+                file_size, checksum, mimetype, index_content
+            FROM ir_attachment
+            WHERE
+                res_model = 'hr.employee'
+                AND res_field = 'image_1920'
+                AND res_id = %s
+            """,
+            (employee_values[0],),
+        )
+        for attachment_values in env.cr.fetchall():
+            openupgrade.logged_query(
+                env.cr,
+                """
+                INSERT INTO ir_attachment (
+                    name, description, res_model, res_field, res_id,
+                    company_id, type, url, public, db_datas, store_fname,
+                    file_size, checksum, mimetype, index_content
+                ) VALUES (
+                    %s, %s, 'res.partner', 'image_1920', %s, %s, %s, %s,
+                    %s, %s, %s, %s, %s, %s, %s
+                )
+                """,
+                (
+                    attachment_values[0],
+                    attachment_values[1],
+                    attachment_values[4],
+                    attachment_values[5],
+                    attachment_values[6],
+                    attachment_values[7],
+                    attachment_values[8],
+                    attachment_values[9],
+                    attachment_values[10],
+                    attachment_values[11],
+                    attachment_values[12],
+                    attachment_values[13],
+                    attachment_values[14],
+                ),
+            )
```

Comment:
This part is new and not tested yet. @robinkeunen can you have a look at this. Is there a way to put all that into one SQL request or any better way to do it ?

Reply:
Robin has no time to suggest better way of doing it. So any though are welcome. :)